### ISSUE TITLE
The `Method` enum should support all the registered HTTP methods

### ISSUE THREAD

**Issue by zenekron** [http-rs/http-types#331]
**TL;DR**: the `Method` enum only supports fraction of all the [registered HTTP methods][Method Registry].

EDIT: I've currently opened the PR #332 to resolve this issue.

### About Mozilla's documentation
Mozilla's [HTTP request methods][MDN http methods] only contains information about the methods specified in [RFC 7231][] and [RFC 5789][]. While these are the most commonly used ones, they only account for 9 out of the 39 (~23%) currently registered HTTP methods.

### What is a *Registered HTTP Method*?
[RFC 7231][] states the following:

> 4  Request Methods
> 4.1  Overview
>
> ...
> 
> Additional methods, outside the scope of this specification, have
> been standardized for use in HTTP.  All such **methods ought to be
> registered within the "Hypertext Transfer Protocol (HTTP) Method
> Registry" maintained by IANA**, as defined in Section 8.1.
> 
> ...
>
> 8  IANA Considerations
> 8.1  Method Registry
> **The "Hypertext Transfer Protocol (HTTP) Method Registry" defines the
> namespace for the request method token (Section 4)**.  The method
> registry has been created and is now maintained at
> <http://www.iana.org/assignments/http-methods>.

Essentially a registered HTTP method is a method that has gone through the process of being added to [IANA's HTTP Method Registry][Method Registry].

[MDN http methods]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods
[RFC 7231]: https://tools.ietf.org/html/rfc7231
[RFC 5789]: https://tools.ietf.org/html/rfc5789
[Method Registry]: https://www.iana.org/assignments/http-methods

**Comment by yoshuawuyts** [comment]
Closes by #332.

**Comment by Mart-Bogdan** [comment]
There should have been open member Other(String) to support further methods, that can be added to IANA later, or ones that can be used privately by applications. Many HTTP servers just use String in place of method

**Comment by Fishrock123** [comment]
I agree
